packages/fx-core/templates/plugins/resource/bot/bicep/botConfiguration.template.bicep

```diff
@@ -8,11 +8,21 @@ param currentAppSettings object
 
 var botWebAppName = split(provisionOutputs.botOutput.value.botWebAppResourceId, '/')[8]
 var m365ClientId = provisionParameters['m365ClientId']
+{{#contains 'fx-resource-key-vault' Plugins}}
+var m365ClientSecret = provisionOutputs.keyVaultOutput.value.m365ClientSecretReference
```

Comment:
expose some references when `generateArmTemplates`, and consume the values here. So other resource plugins do not need to hardcode the reference values here. There's a `Reference` property in the result type of `generateArmTemplates` function.